src/ocamlobjinfo.mll

```diff
@@ -0,0 +1,41 @@
+{
+open Stdune
+
+type t = Module.Name.Set.t Ml_kind.Dict.t
+
+let empty =
+  { Ml_kind.Dict.
+    intf = Module.Name.Set.empty
+  ; impl = Module.Name.Set.empty
+  }
+
+let add_intf (t : t) i =
+  { t with intf = Module.Name.Set.add t.intf (Module.Name.of_string i) }
+let add_impl (t : t) i =
+  { t with impl = Module.Name.Set.add t.impl (Module.Name.of_string i) }
+}
+
+let newline = '\r'? '\n'
+let ws = [' ' '\t']+
+let hash = ['0'-'9' 'a'-'z']+
+let name = ['A'-'Z'] ['a'-'z' '0'-'9' '_']*
+
+rule ocamlobjinfo acc = parse
```

Comment:
How about exposing this "pure" part in the .mli and writing a test for it? It doesn't have to be fancy but I think it'd be more comfortable with a (small) safety net.